src/defs.h

```diff
@@ -438,18 +477,28 @@ extern TicksTime g_cycleStartTime;
 
 extern debug_level_t g_debug_level;
 
-#ifdef USING_VMA_EXTRA_API
-
+#if defined(USING_VMA_EXTRA_API) || defined(USING_XLIO_EXTRA_API)
+#ifdef USING_VMA_EXTRA_API // VMA
 extern struct vma_buff_t *g_vma_buff;
 extern struct vma_completion_t *g_vma_comps;
-
+#endif
 class ZeroCopyData {
 public:
     ZeroCopyData();
     void allocate();
     ~ZeroCopyData();
     unsigned char *m_pkt_buf;
-    struct vma_packets_t *m_pkts;
+    union {
+#ifdef USING_VMA_EXTRA_API // VMA packet
+        struct vma_packets_t *m_pkts_vma;
+#endif
+#ifdef USING_XLIO_EXTRA_API // XLIO packet
+        struct xlio_recvfrom_zcopy_packets_t *m_pkts_xlio;
+#endif
```

Comment:
It is not clear why above are needed. All processing can be done using `void *m_pkts`

Reply:
Please consider casting in code. I think it is more visible.

Reply:
Done